PocketKit/Sources/Sync/CoreDataInitializers/SavedItem+CoreDataProperties.swift

```diff
@@ -13,14 +13,14 @@ extension SavedItem {
     }
 
     @NSManaged public var archivedAt: Date?
-    @NSManaged public var createdAt: Date?
+    @NSManaged public var createdAt: Date
```

Comment:
making this non-optional causes our tests for `space.buildPendingSavedItem()` to crash, can we conform that pending saved item will definitely have a `createdAt` field?

`test_selectCell_whenItemIsArticle_setsSelectedItemToReaderView`

Reply:
strictly speaking, `SavedItem.createdAt` is not optional on the graph, so if it does not have any side effect, we can just set that date. On the other hand, if we don't directly use that field and the backend would set the value to the item when it's created (on the backend), we might also keep that optional. @bassrock what do you think?

Reply:
Lets set the date ourselves and use the rule that server state always takes priority.

Reply:
Should we add `createdAt` to the `init` for the SavedItem?  Also, are we okay having `remoteID` as `""`?
![image](https://user-images.githubusercontent.com/6743397/233134067-8b0f46f4-2088-466d-9001-361e7ec99618.png)

`PocketKit/Sources/Sync/CoreDataInitializers/SavedItem+CoreDataClass.swift`

Reply:
> Should we add createdAt to the init for the SavedItem?
we can potentially use the local date of creation, not totally sure if the backend will use this or not
>  are we okay having remoteID as `""` ?
I don't think that's an option, because that needs to be a unique ID, unless the backend handles empty remoteIDs by creating a proper one?
@bassrock would you have some more insights on the above?